src/python.rs

```diff
@@ -556,17 +556,42 @@ impl<'p> Python<'p> {
         FromPyPointer::from_borrowed_ptr_or_opt(self, ptr)
     }
 
-    /// Lets the Python interpreter check for pending signals and invoke the
-    /// corresponding signal handlers. This can run arbitrary Python code.
+    /// Lets the Python interpreter check and handle any pending signals and invoke the
+    /// corresponding signal handlers.
```

Comment:
There's quite a few "and" in this sentence now, maybe the below is slightly clearer?

```suggestion
    /// Lets the Python interpreter check and handle any pending signals. This will invoke the
    /// corresponding signal handlers registered in Python (if any).
```